## Úloha 1

In [1]:
import math #potrebne pre funkciu math.log
x = 14 + (1/2)**math.log(15)
print(x)

14.153036723776381


X sa rovná 14.153036723776381.

## Úloha 2

In [3]:
n = 1/3 * math.pi
print(n)
y = math.cos(n)
print(y)

1.0471975511965976
0.5000000000000001


Y sa rovná 0.5000000000000001.

## Úloha 3

In [4]:
z = 1 - n**2/math.factorial(2) + n**4/math.factorial(4) - n**6/math.factorial(6) + n**8/math.factorial(8)
print(z)

0.500000433432915


Z sa rovná 0.500000433432915.

## Úloha 4
Vypočitaj absolutnu hodnotu rozdielu |z − y| a výsledok zaukrúhli na 3 
desatinné miest.1

In [5]:
a = abs(z - y) #abs nie je súčasťou packagu math
print(a)
print(round(a,3)) #round(zaokrúhľované číslo/premenná, počet desatinných miest)

4.3343291489250646e-07
0.0


## Úloha 5 
Vypočitaj zvyšok po delení 5/2, ,
/22
7

In [6]:
print ('Zvyšok po delení:',5%2,',', 22%7)

Zvyšok po delení: 1 , 1


## Úloha 6
. Otestuj nasledujúce:
a = 15
b = 15
a = a + 15
b +=15
čo vykonal symbol += ? Ide o tzv. "assignment operator" vid napr.
https://www.programiz.com/python-programming/operators časť 2
napíš ekvivalenty k nasledujúceim príkladom :
a = a−5
a = a∗5
a = a ∗∗5

In [7]:
a = 15
b = 15
a = a + 15
b += 15 # znak '+=' pripočíta hodnotu, a zároveň priradí túto novú hodnotu premennej v jednom kroku
print(a, b)

#zjednodušené varianty uvedených príkladov:
a -= 5
print(a)
a *= 5
print(a)
a**= 5
print(a)

30 30
25
125
30517578125


## Úloha 7
7. Otestuj funkciou type() na premenje x a y ak x = 15 a y = 15.0

In [4]:
x = 15
y = 15.0
print(type(x))
print(type(y))

<class 'int'>
<class 'float'>


## Úloha 8
8. Nasledujúce čísla deklaruj pomocou scientific notation2
x = 1587.1015 (4)
x = 859762.56 = 8.5976256.105

In [8]:
x = 1.5871014e17 #scientific notation formát: číslo v skrátenej podobe + znak 'e' + rád (order of magnitude)
print(x)
x = 8.5976256e5
print(x)

1.5871014e+17
859762.56


## Úloha 9
Importuj z knižnice numpy funkciu na generovanie náhodného čísla a
implementuj ju. čo je to seed ?
3

In [8]:
from numpy import random
j = random.randint(100)
print(j)

73


Seed je input pre generovanie pseudorandom čísel. Seed robí random generátor predikovateľným.

## Úloha 10
Je možné aby integer v pythnone pretiekol ?
-áno